sequential_solver/solver/solver.py

```diff
@@ -0,0 +1,1326 @@
+import itertools
```

Comment:
This file is rather large, so I would suggest to split it logically, to make debugging or future improvements easier in the future.